lib/elixir/lib/calendar/time.ex

```diff
@@ -418,7 +434,7 @@ defmodule Time do
       -2_000_000
 
   """
-  @spec diff(Time.t, Time.t, System.time_unit) :: integer
+  @spec diff(t, t, System.time_unit) :: integer
```

Comment:
yes, I thought about it, but it's potentially error-prone: `Time.diff(~N[2000-01-01 09:00:00], ~N[1970-01-01 08:00:00])` will still return just 3600 seconds, so it _might_ be a subtle bug in someone's code. But it's a nice feature as well. That's why I wasn't sure.